DiamondLightSource/SynchWeb
# LIMS-656: Show auto subsamples if they have data

**JIRA ticket**: [LIMS-656](https://jira.diamond.ac.uk/browse/LIMS-656)

**Summary**:

If a subsample on a plate has an associated data collection, then it should always be shown, even if it is from an old inspection.

**Changes**:
- Add an optional extra UNION to the subsample query showing "auto" samples with data collections

**To test**:
- Go to /containers/cid/249257 and click on the bottom left most well (well 169). There should be (at time of writing) 4 extra subsamples in the list under "Marked Subsamples", the first has X=2286, Y=716 and corresponds to /dc/visit/nt30330-87/id/10216910
![image](https://github.com/user-attachments/assets/4bd40b8d-f9e8-4157-9876-76fe7f0ad326)

- Check well 1 on the same plate, there should be no difference in the number of subsamples shown as none have any data
